lib/processPbrMetallicRoughness.js

```diff
@@ -0,0 +1,774 @@
+                fragmentShaderMain += '  mat3 tbn = mat3(t, b, ng);\n';
+                fragmentShaderMain += '  vec3 n = texture2D(u_normalTexture, ' + v_texcoord + ').rgb;\n';
+                fragmentShaderMain += '  n = normalize(tbn * (2.0 * n - 1.0));\n';
+            } else {
+                fragmentShaderMain += '  vec3 n = ng;\n';
+            }
+            if (parameterValues.doubleSided) {
+                fragmentShaderMain += '  if (!gl_FrontFacing)\n';
+                fragmentShaderMain += '  {\n';
+                fragmentShaderMain += '    n = -n;\n';
+                fragmentShaderMain += '  }\n';
+            }
+        }
+
+        // Add base color to fragment shader
+        if (defined(parameterValues.baseColorTexture)) {
+            fragmentShaderMain += '  vec3 baseColor = texture2D(u_baseColorTexture, ' + v_texcoord + ').rgb;\n';
+            if (defined(parameterValues.baseColorFactor)) {
+                fragmentShaderMain += '  baseColor *= u_baseColorFactor.rgb;\n';
+            }
+        } else {
+            if (defined(parameterValues.baseColorFactor)) {
+                fragmentShaderMain += '  vec3 baseColor = u_baseColorFactor.rgb;\n';
+            } else {
+                fragmentShaderMain += '  vec3 baseColor = vec3(1.0);\n';
+            }
+        }
+        // Add metallic-roughness to fragment shader
+        if (defined(parameterValues.metallicRoughnessTexture)) {
+            fragmentShaderMain += '  vec3 metallicRoughness = texture2D(u_metallicRoughnessTexture, ' + v_texcoord + ').rgb;\n';
+            fragmentShaderMain += '  float metalness = clamp(metallicRoughness.b, 0.0, 1.0);\n';
+            fragmentShaderMain += '  float roughness = clamp(metallicRoughness.g, 0.04, 1.0);\n';
+            if (defined(parameterValues.metallicFactor)) {
+                fragmentShaderMain += '  metalness *= u_metallicFactor;\n';
+            }
+            if (defined(parameterValues.roughnessFactor)) {
+                fragmentShaderMain += '  roughness *= u_roughnessFactor;\n';
+            }
+        } else {
+            if (defined(parameterValues.metallicFactor)) {
+                fragmentShaderMain += '  float metalness = clamp(u_metallicFactor, 0.0, 1.0);\n';
+            } else {
+                fragmentShaderMain += '  float metalness = 1.0;\n';
+            }
+            if (defined(parameterValues.roughnessFactor)) {
+                fragmentShaderMain += '  float roughness = clamp(u_roughnessFactor, 0.04, 1.0);\n';
+            } else {
+                fragmentShaderMain += '  float roughness = 1.0;\n';
+            }
+        }
+        fragmentShaderMain += '  vec3 v = -normalize(v_positionEC);\n';
+
+        // Generate fragment shader's lighting block
+        var fragmentLightingBlock = '';
+        fragmentLightingBlock += '  vec3 lightColor = vec3(1.0, 1.0, 1.0);\n';
+        fragmentLightingBlock += '  vec3 l = normalize(czm_sunDirectionEC);\n';
```

Comment:
Using `czm_sunDirectionEC` should only happen if `optimizeForCesium` is true. Otherwise the light direction should originate from the camera (0.0, 0.0, 1.0) like how `processModelMaterialsCommon` handles it.